tests/ops/qutrit/test_qutrit_non_parametric_ops.py

```diff
@@ -0,0 +1,122 @@
+        copied_op = copy.copy(op)
+        np.testing.assert_allclose(op.matrix(), copied_op.matrix(), atol=tol)
+
+        op._inverse = True
+        copied_op2 = copy.copy(op)
+        np.testing.assert_allclose(op.matrix(), copied_op2.matrix(), atol=tol)
+
+    @pytest.mark.parametrize("ops, mat", NON_PARAMETRIZED_OPERATIONS)
+    def test_matrices(self, ops, mat, tol):
+        """Test matrices of non-parametrized operations are correct"""
+        op = ops(wires=range(ops.num_wires))
+        res_static = op.compute_matrix()
+        res_dynamic = op.matrix()
+        assert np.allclose(res_static, mat, atol=tol, rtol=0)
+        assert np.allclose(res_dynamic, mat, atol=tol, rtol=0)
+
+
+class TestEigenval:
+    def test_tshift_eigenval(self):
+        """Tests that the TShift eigenvalue matches the numpy eigenvalues of the TShift matrix"""
+        op = qml.TShift(wires=0)
+        exp = np.linalg.eigvals(op.matrix())
+        res = op.eigvals()
+        assert np.allclose(res, exp)
+
+    def test_tclock_eigenval(self):
+        """Tests that the TClock eigenvalue matches the numpy eigenvalues of the TClock matrix"""
+        op = qml.TClock(wires=0)
+        exp = np.linalg.eigvals(op.matrix())
+        res = op.eigvals()
+        assert np.allclose(res, exp)
+
+
+period_three_ops = (
+    qml.TShift(wires=0),
+    qml.TClock(wires=0),
+)
+
+
+class TestPowMethod:
+    @pytest.mark.parametrize("op", period_three_ops)
+    @pytest.mark.parametrize("n", (-5, -2, 1, 4, 7))
+    def test_period_three_1_mod_3(self, op, n):
+        """Tests that ops raised to an integer power == 1 mod 3 are the same as the original op"""
+        assert op.pow(n)[0].__class__ is op.__class__
+
+    @pytest.mark.parametrize("op", period_three_ops)
+    @pytest.mark.parametrize("n", (-4, -1, 2, 5, 8))
+    def test_period_three_2_mod_3(self, op, n):
+        """Tests that ops raised to an integer power == 2 mod 3 are the adjoint of the original op"""
+        op_pow = op.pow(n)[0]
+        assert op_pow.__class__ is op.__class__
+        assert np.allclose(op.matrix().conj().T, op_pow.matrix())
+        assert op_pow.inverse == True
+
+    @pytest.mark.parametrize("op", period_three_ops)
+    @pytest.mark.parametrize("n", (-6, -3, 0, 3, 6))
+    def test_period_three_0_mod_3(self, op, n):
+        """Tests that ops raised to an integer power == 0 mod 3 are empty lists"""
+        assert len(op.pow(n)) == 0
+
+    @pytest.mark.parametrize("op", period_three_ops)
+    def test_period_three_noninteger_power(self, op):
+        """Test that ops with a period of 3 raised to a non-integer power raise an error."""
+        with pytest.raises(qml.operation.PowUndefinedError):
+            op.pow(1.234)
+
+
+label_data = [
+    (qml.TShift(0), "TShift", "TShift⁻¹"),
+    (qml.TClock(0), "TClock", "TClock⁻¹"),
+]
+
+
+@pytest.mark.parametrize("op, label1, label2", label_data)
+def test_label_method(op, label1, label2):
+    assert op.label() == label1
```

Comment:
How are these passing? I didn't notice labels in the non-parametric ops file (are they defined somewhere else and I missed them?)

Reply:
The base label method is defined in `pennylane/operation.py`, and it uses the class name to get the base label and adds the inverse symbol if the operation is adjointed, which is why `label` didn't need to be overriden. It would however be overridden for methods that are self-adjoint, such as the `TSWAP` operation.